Components: Refactor `Guide` tests to `@testing-library/react`

tyxla opened the pull request:
## What?
We've recently started refactoring `enzyme` tests to `@testing-library/react`.

This PR refactors the `<Guide />` component tests from `enzyme` to `@testing-library/react`.

## Why?
`@testing-library/react` provides a better way to write tests for accessible components that is closer to the way the user experiences them.

## How?
We're straightforwardly replacing `enzyme` tests with `@testing-library/react` ones, using `jest-dom` matchers screen queries.

This also uncovers a bug: when we render `<Guide />` with no pages, we attempt to focus the guide container, however it doesn't exist because we don't render it if no pages exist. This PR fixes that bug by changing the component to focus only if the guide container exists.

## Testing Instructions
* Verify tests pass: `npm run test:unit packages/components/src/guide`
* Verify Guide component still works well in Storybook.

tyxla committed:
Components: Refactor Guide tests to @testing-library/react

tyxla committed:
Add changelog

ciampo commented on packages/components/src/guide/test/index.js:
```diff
@@ -1,104 +1,138 @@
+		 *
+		 * because the event sent has a `keyCode` of `0`.
+		 *
+		 * To fix this, we'll need to update the Modal component to work with `KeyboardEvent.code`.
```
Yeah, this same issue was encountered by @chad1008 when working on some other unit tests 😅 

Opened #43429 so that we can remove `fireEvent` completely from this PR before merging

ciampo reviewed (commented):
Love seeing this sort of refactors, the tests have gained a lot more value!

> This also uncovers a bug: when we render <Guide /> with no pages, we attempt to focus the guide container, however it doesn't exist because we don't render it if no pages exist. This PR fixes that bug by changing the component to focus only if the guide container exists.

Amazing!

tyxla committed:
fireEvent -> user-event

tyxla commented:
> Yeah, this same issue was encountered by @chad1008 when working on some other unit tests 😅
> 
> Opened #43429 so that we can remove `fireEvent` completely from this PR before merging

Thanks a bunch, @ciampo! I've rebased after it landed and updated the tests to use `user-event` now. 

This is now ready for another look.

ciampo commented on packages/components/CHANGELOG.md:
```diff
@@ -37,10 +38,12 @@
 
 ### Experimental
 -   `FormTokenField`: add `__experimentalAutoSelectFirstMatch` prop to auto select the first matching suggestion on typing ([#42527](https://github.com/WordPress/gutenberg/pull/42527/)).
+=======
```
Probably a survivor of a resolved merge conflict :) 

```suggestion
```

tyxla replied:
Oops, great catch, thanks 💯

ciampo reviewed (approved):
LGTM 🚀

tyxla committed:
Clean up after bad rebase

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>
